caching.py

```
@@ -0,0 +1,613 @@
+            self.redis_client.flushdb()
+            self.stats.size = 0
+            self._update_stats()
+        except Exception as e:
+            logger.error(f"Redis clear error: {e}")
+    
+    def get_stats(self) -> CacheStats:
+        """Get cache statistics."""
+        try:
+            info = self.redis_client.info()
+            self.stats.size = info.get('db0', {}).get('keys', 0)
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)